### ISSUE TITLE
switch from net2 to socket2

### ISSUE THREAD

**Issue by kallisti5** [tokio-rs/mio#1319]
net2 is now deprecated.   mio should switch to socket2 which is supported.

**Comment by Thomasdezeeuw** [comment]
Mio doesn't use net2 anymore.

**Comment by kallisti5** [comment]
https://github.com/tokio-rs/mio/blob/40df934a11b05233a7796c4de19a4ee06bc4e03e/Cargo.toml#L48 ??

**Comment by Thomasdezeeuw** [comment]
I know its only used in a single test: https://github.com/tokio-rs/mio/blob/0ba2ed8bb1e3af999024653d587b9c25f8d4b57c/tests/tcp.rs#L472

So its not really a high priority.

**Comment by kallisti5** [comment]
Ok. So all of net2 which is broken on some platforms gets pulled in and compiled by every crate which depends on tokio and mio for a single test.

**Comment by Thomasdezeeuw** [comment]
No it’s a development dependency which means it’s only used when run ‘cargo test’ in the Mio crate

**Comment by kallisti5** [comment]
weird... the Cargo.lock tells another story.

```
[[package]]
name = "mio"
version = "0.6.22"
source = "registry+https://github.com/rust-lang/crates.io-index"
dependencies = [
 "cfg-if 0.1.10 (registry+https://github.com/rust-lang/crates.io-index)",
 "fuchsia-zircon 0.3.3 (registry+https://github.com/rust-lang/crates.io-index)",
 "fuchsia-zircon-sys 0.3.3 (registry+https://github.com/rust-lang/crates.io-index)",
 "iovec 0.1.4 (registry+https://github.com/rust-lang/crates.io-index)",
 "kernel32-sys 0.2.2 (registry+https://github.com/rust-lang/crates.io-index)",
 "libc 0.2.69 (registry+https://github.com/rust-lang/crates.io-index)",
 "log 0.4.8 (registry+https://github.com/rust-lang/crates.io-index)",
 "miow 0.2.1 (registry+https://github.com/rust-lang/crates.io-index)",
 "net2 0.2.34 (registry+https://github.com/rust-lang/crates.io-index)",
 "slab 0.4.2 (registry+https://github.com/rust-lang/crates.io-index)",
 "winapi 0.2.8 (registry+https://github.com/rust-lang/crates.io-index)",
]
```

**Comment by Keruspe** [comment]
@kallisti5 That's mio 0.6, latest (and fixed) version is 0.7

**Comment by kallisti5** [comment]
ah ha!  Installing cargo-tree shows it's actually coming from the old log dependency.  Any chance of a version bump from 0.4.8 to 0.4.10?  (or maybe 0.4.x)

**Comment by Veetaha** [comment]
`cargo-deny` reports `net2` unmaintained advisory even for `dev-dependencies` of any crate in the tree.
It's unfortunate having to add an `ignore` clause for it, I think many people do stumble with this just adding `tokio` as a dependency.
@Thomasdezeeuw, could you please reevaluate this decision and maybe remove/update the test in favor of not making the ecosystem stumble on the advisory?

**Comment by taiki-e** [comment]
> `cargo-deny` reports `net2` unmaintained advisory even for `dev-dependencies` of any crate in the tree.

No, that is due to tokio(0.2) depends on old mio(0.6). Updating tokio to use the latest mio(0.7) that fixed this issue is a breaking change, and this should be fixed in tokio 0.3 (unreleased).

Another solution to that problem is to backport the removal of net2 dependency to mio 0.6 but I'm not sure if that is possible without breaking changes.

**Comment by Thomasdezeeuw** [comment]
@Veetaha like @taiki-e pointed out Tokio is the problem here. The plan was to port Tokio to Mio v0.7 in Q2 of 2020, but well we all know what happened and so far it hasn't happened yet. So I see two ways going forward:
1) Port Tokio to Mio v0.7, this is a breaking change in Tokio (as it exposes Mio traits), so that would become Tokio v0.3.
2) As @taiki-e mentioned backport the removal of net2 to Mio v0.6. I'm open to this, but not open to switch to socket2 as that crate is simply not production ready.
